assets/js/segment.js

```diff
@@ -0,0 +1,79 @@
+window.addEventListener('DOMContentLoaded', function () {
+  // If target element is code copy
+  function isTargetCodeCopyButton(elem) {
+    return elem.classList.contains('fa-copy');
+  }
+
+  // Determine whether we should track or not
+  function shouldTrack(e) {
+    if (e.target.tagName === 'A' || isTargetCodeCopyButton(e.target)) {
+      return { target: e.target };
+    } else if (e.target.parentNode.tagName === 'A') {
+      // For image/icon
+      return { target: e.target, parent: e.target.parentNode };
+    }
+
+    // Not tracking this
+    return null;
+  }
+
+  // Get the payload to be sent
+  function getPayload({ target, parent }) {
+    let hrefUrl;
+    let text;
+
+    if (parent) {
+      hrefUrl = parent.href;
+
+      if (target.classList.contains('logo')) {
+        // Logo click
+        text = 'logo';
+      } else {
+        text = target.getAttribute('title') || target.getAttribute('alt');
+      }
+    } else if (isTargetCodeCopyButton(target)) {
+      // Code copy
+      // @TODO: No spec yet
+      return;
+    } else {
+      hrefUrl = target.href;
+      text = target.textContent;
+    }
+
+    return {
+      hrefUrl,
+      text: (text || '').trim(),
+    };
+  }
+
+  document.body.addEventListener('click', function (e) {
```

Comment:
Are you tracking all clicks, regardless of whether or not it is a link? If you track clicks through bubbling then this will also capture clicks on blank sections of the page. Not sure if that matters or not

Reply:
Yes, using bubbling, we can track all clicks regardless of DOM hierarchy. I do have additional checks on whether or not to send the events.